src/app/user/user-details/user-details.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable, of, Subject } from 'rxjs';
+import { User } from '../../shared/models/user.model';
+import { UserService } from '../../shared/services/user.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { takeUntil, catchError } from 'rxjs';
+import { ParamMap } from '@angular/router';
+import { AsyncPipe } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import {MatCheckboxModule} from '@angular/material/checkbox';
+import { ReactiveFormsModule, FormControl, FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'user-details',
+  standalone: true,
+  imports: [AsyncPipe, MatIconModule, MatButtonModule, MatTooltipModule, MatCheckboxModule, ReactiveFormsModule],
+  templateUrl: './user-details.component.html',
+  styleUrl: './user-details.component.scss'
+})
+export class UserDetailsComponent implements OnInit, OnDestroy {
+  user$: Observable<User> | null = null;
+  private destroy$: Subject<void> = new Subject<void>();
+  editable: boolean = false;
+
+  roleForm = new FormGroup({
+    user: new FormControl({ value: false, disabled: true }),
```

Comment:
you should assign the correct values (true or false) based on the user response, so once you get the user details you can use rxjs operator tap to check for roles and update the formGroup accordingly 🙂

Reply:
that way I think you also won't have to declare a roles array with an initial value of User, also User should be disabled in the template, as it should always be a role a user has (I don't remember if I wrote that in the ticket, if I didn't apologies)